mimic/model/identity_objects.py

```diff
@@ -62,17 +80,164 @@ def not_found(message, request):
+    return _identity_error_message("unauthorized", message, UNAUTHORIZED, request)
+
+
+@implementer(IEndpointTemplate, IPlugin)
+class EndpointTemplateStore(object):
+    """
+    A :obj"`EndpointTemplateStore` is an object whiich provides the
+    functionality to integrate an Endpoint Template instance. It is only
+    meant to work with a single Endpoint Template. Use :obj:`ExternalApiStore`
+    to manage multiple Endpoint Templates.
+
+    :ivar list required_mapping: list of entries in the template that are
+        required to be present for a valid template
+    :ivar list optional_mapping: list of entries in the template that are
+        optionally present in a valid template, along with a default value.
+
+
+    Note: The OpenStack documentation[0] does not specify any required
+        parameters. For this implementation, the `id`, `region`, `type`,
+        and `name` fields are required. The `name` and `type` fields
+        are used for creating an instance of the :obj:`ExternalApiStore`
+        that will enable the listing to be put into the Service Catalog;
+        the `id` and `region` fields are used by the :obj:`ExternalApiStore`
+        for managing the templates.
+
+    [0] http://developer.openstack.org/api-ref-identity-v2-ext.html
+    """
+
+    """
+    The following lists of tuples are for mapping the template defined by
+    the OpenStack documentation to the internal strucutre for several uses.
+    The tuples have the following meaning:
+
+        Index   Meaning
+        -----   ------------------------------------------------
+        0       JSON Key in the OpenStack Identity Template Spec
+        1       Attribute Name of the :obj:`EndpointTemplateStore`
+        2       default value (only for optional_mapping)
+    """
+    required_mapping = [
+        ('id', 'id_key'),
+        ('region', 'region_key'),
+        ('type', 'type_key'),
+        ('name', 'name_key'),
+    ]
+
+    optional_mapping = [
+        ('enabled', 'enabled_key', False),
+        ('publicURL', 'public_url', u""),
+        ('internalURL', 'internal_url', u""),
+        ('adminURL', 'admin_url', u""),
+        ('RAX-AUTH:tenantAlias', 'tenant_alias', "%tenant_id%"),
+        ('versionId', 'version_id', u""),
+        ('versionInfo', 'version_info', u""),
+        ('versionList', 'version_list', u"")
+    ]
+
+    tenant_mapping = [
+        ('id', 'id_key'),
+        ('region', 'region_key'),
+        ('type', 'type_key'),
+        ('publicURL', 'public_url'),
+        ('internalURL', 'internal_url'),
+        ('adminURL', 'admin_url'),
+    ]
+
+    def __init__(self, template_dict=None):
+        """
+        Create an :obj:`EndpointTemplateStore`
+
+        :param dict template_dict (optional): optional dictionary to load the
+            Endpoint Template data from. The dictionary is simply passed to the
+            deserializer.
+        """
+        self._template_data = template_dict
```

Comment:
This could be made a lot shorter / simpler / safer / consistent by being an `attr.s`-ified class.